ca_tools/scripts/draw_square.py

```diff
@@ -0,0 +1,209 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+import math
+from GTSys import GTSys
+from RobotLocalizationTf import RobotLocalizationTf
+
+import rospy
+from geometry_msgs.msg import Twist, Pose, Quaternion, PoseWithCovariance
+from nav_msgs.msg import Odometry
+from std_msgs.msg import Bool
+from tf.transformations import euler_from_quaternion as efq
+from tf.transformations import quaternion_from_euler as qfe
+
+
+
+class RobotState():
+
+    STATES = [
+    "MOVE_FORWARD",
+    "ROTATE",
+    "STOP"
+    ]
+
+    _current_state = None
+
+    def __init__(self):
+        self._current_state = "STOP"
+
+    def SetState(self,state):
+        "ToDo: assert a valid state"
```

Comment:
An [enum](https://docs.python.org/3/library/enum.html) might be useful.

Reply:
Didn't know that this exists in Python. Thanks!